src/sink.ts

```diff
@@ -325,11 +329,18 @@ class Sink {
    */
   async setMetadata(metadata: SetSinkMetadata): Promise<SinkMetadataResponse> {
     const [currentMetadata] = await this.getMetadata();
-    const reqOpts = {
+    const uniqueWriterIdentity = metadata.uniqueWriterIdentity;
+    delete metadata.uniqueWriterIdentity;
+    let reqOpts = {
       sinkName: this.formattedName_,
       sink: extend({}, currentMetadata, metadata),
     };
     delete reqOpts.sink.gaxOptions;
+    // Add user specified uniqueWriterIdentity boolean, if any.
+    reqOpts = {
+      ...reqOpts,
+      ...(uniqueWriterIdentity && {uniqueWriterIdentity}),
+    };
```

Comment:
We could just do:

```js
if (metadata.uniqueWriterIdentity) {
  reqOpts.uniqueWriterIdentity = metadata.uniqueWriterIdentity;
}
```

Reply:
Doing that yields typescript error `TS2339: Property 'uniqueWriterIdentity' does not exist on type '{ sinkName: string; sink: ILogSink & SetSinkMetadata; }'.`

Would prefer to not have to formally define reqOpts since we're transiently using it in this function